File: Recursion/JS/factorial.js
//TODO: Find the factorial of a given number

function factorial(x) {

    // if number is 0
    if (x == 0) {                             //TODO: Base statement
        return 1;
    }

    // if number is positive
    else {
        return x * factorial(x - 1);          //TODO: Recursive statement
    }
}

                                              //TODO: take input from the user
let num = prompt('Enter a positive number: ');

                                              //TODO:calling factorial() if num is positive
if (num >= 0) {
    let result = factorial(num);
    console.log(`The factorial of ${num} is ${result}`);    //TODO: Print result
}
else {
    console.log('Enter a positive number.');
}

File: Recursion/C++/factorial.cpp
//TODO: Print factorial of the given number

#include<iostream>
using namespace std;

int factorial(int n);

int main()
{
    int n;

    cout << "Enter a positive integer: ";                         //TODO: Getting user input
    cin >> n;

    cout << "Factorial of " << n << " = " << factorial(n);        //TODO: Printing result

    return 0;
}

int factorial(int n)
{
    if(n > 1)
        return n * factorial(n - 1);                              //TODO: Recursive statement
    else
        return 1;                                                 //TODO: Base statement
}
